src/audio/dai-zephyr.c

```diff
@@ -143,6 +143,9 @@ int dai_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	case SOF_DAI_INTEL_SSP:
 		cfg.type = is_blob ? DAI_INTEL_SSP_NHLT : DAI_INTEL_SSP;
 		cfg_params = is_blob ? spec_config : &sof_cfg->ssp;
+		if (dai_set_link_hda_config(&cfg.link_config,
+					    common_config, cfg_params) < 0)
+			return -EINVAL;
```

Comment:
since `dai_set_link_hda_config()` cannot fail, not much reason to have it return an int and not much point of checking the return value?